nova/compute/manager.py

```diff
@@ -10680,6 +10681,35 @@ def _reclaim_queued_deletes(self, context):
                                 "instance: %s",
                                 e, instance=instance)
 
+    @periodic_task.periodic_task(
+            spacing=CONF.instances_stuck_in_deleting_cleanup_interval)
+    def _cleanup_instances_stuck_in_deleting(self, context):
+        """Reclaim instances stuck in DELETING state, but not really destroyed,
+        and retrigger terminate_instance for them.
+        """
+        filters = {'task_state': task_states.DELETING,
+                   'host': self.host}
+        instances = objects.InstanceList.get_by_filters(
+            context, filters,
+            expected_attrs=objects.instance.INSTANCE_DEFAULT_FIELDS,
+            use_slave=True)
+        for instance in instances:
+            # Check if the instance has a process lock
+            for name, s in _semaphores._semaphores.items():
+                if name == instance.uuid:
+                    break
+            else:
```

Comment:
This looks needlessly compllicated. Should it be the same as this?
```python
if instance.uuid in lockutiils._semaphores._semaphores:
    continue
```

Reply:
fixed